ForumSessionProvider/extension.json

```diff
@@ -46,7 +46,7 @@
   },
   "AutoloadClasses": {
     "ForumSessionProvider": "ForumSessionProvider.php",
-    "AuthManager_SMF": "ForumSessionProvider.php"
+    "AuthManager_ForumSoftware": "ForumSessionProvider.php"

```

Comment:
For naming consistences, I'd suggest ForumAuthManager, and again make it a separate file.